dwave/optimization/include/dwave-optimization/nodes/mathematical.hpp

```diff
@@ -37,6 +37,11 @@ struct logical {
     constexpr bool operator()(const T& x) const { return x; }
 };
 
+template <class T>
+struct logical_xor {
+    constexpr bool operator()(const T& x, const T& y) const { return static_cast<bool>(x)^static_cast<bool>(y); }
+};
```

Comment:
A bit of formatting for ling length. Also, I am sure the bitwise operator is OK, but just using `!=` is probably a bit safer.
```suggestion
template <class T>
struct logical_xor {
    constexpr bool operator()(const T& x, const T& y) const {
        return static_cast<bool>(x) != static_cast<bool>(y);
    }
};
```